src/main/java/com/google/vinet/servlets/RequestServlet.java

```diff
@@ -0,0 +1,268 @@
+  /** The {@code UserService} implementation that this {@code RequestServlet depends on}. */
+  private final UserService userService;
+  /** The {@code RegistrationServlet} implementation that this {@code RequestServlet depends on}. */
+  private final RegistrationServlet registrationServlet;
+
+  /**
+   * Construct a RequestServlet with all of its dependencies set to their default implementations.
+   */
+  public RequestServlet() {
+    this.datastore = DatastoreServiceFactory.getDatastoreService();
+    this.userService = UserServiceFactory.getUserService();
+    this.registrationServlet = new RegistrationServlet();
+  }
+
+  /**
+   * Construct a RequestServlet which depends on the provided dependencies.
+   * @param datastore The DatastoreService implementation to depend on.
+   * @param userService The UserService implementation to depend on.
+   * @param registrationServlet The RegistrationServlet implementation to depend on.
+   */
+  public RequestServlet(DatastoreService datastore, UserService userService, RegistrationServlet registrationServlet) {
+    this.datastore = datastore;
+    this.userService = userService;
+    this.registrationServlet = registrationServlet;
+  }
+
+  /**
+   * Post an Isolate's request to the servlet. Both the request and its tickets will be put into the DataStore.
+   * @param request The request to be read.
+   * @param response The response to be written to.
+   * @throws IOException If an IOException occurs while reading from the request or writing to the reponse.
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response)
+      throws IOException {
+    if (response == null) {
+      throw new IllegalArgumentException("response must not be null");
+    }
+
+    if (request == null) {
+      throw new IllegalArgumentException("request must not be null");
+    }
+
+    if (!this.userService.isUserLoggedIn()) {
+      response.sendError(
+          HttpServletResponse.SC_UNAUTHORIZED,
+          "user must be logged in to post a request"
+      );
+      return;
+    }
+
+    final boolean registered = registrationServlet.isUserRegistered();
+
+    if (!registered) {
+      response.sendError(
+          HttpServletResponse.SC_UNAUTHORIZED,
+          "user must be registered to post a request"
+      );
+      return;
+    }
+
+    final boolean isIsolate = registrationServlet.isUserIsolate();
+
+    if (!isIsolate) {
+      response.sendError(
+          HttpServletResponse.SC_UNAUTHORIZED,
+          "user must be registered as an isolate to post a request"
+      );
+      return;
+    }
+
+    String date = request.getParameter("date");
+    String duration = request.getParameter("duration");
+    String startTime = request.getParameter("startTime");
+    String endTime = request.getParameter("endTime");
+    String timezone = request.getParameter("timezoneId");
+    String[] subjectsRAW = request.getParameterValues("subject");
```

Comment:
nit: can you add a comment that subjects and details are matched based on their index in the array.

Reply:
I've included this now https://github.com/googleinterns/step226-2020/pull/8/commits/3a78b1d88b76860fed697c5345edefa52fdc5802